packages/compass-aggregations/src/stores/store.js

```diff
@@ -185,12 +44,57 @@ export const setSourceNames = (store, sourceName) => {
  * @returns {Store} The store.
  */
 const configureStore = (options = {}) => {
-  const store = createStore(reducer, applyMiddleware(thunk));
+  const { collection } = toNS(options?.namespace ?? '');
+
+  const store = createStore(
+    reducer,
+    {
+      appRegistry: {
+        localAppRegistry: options.localAppRegistry ?? null,
+        globalAppRegistry: options.globalAppRegistry ?? null
+      },
+      dataService: {
+        error: options.dataProvider?.error ?? null,
+        dataService: options.dataProvider?.dataProvider ?? null
+      },
+      namespace: collection ? options.namespace : undefined,
```

Comment:
This doesn't really makes sense to me, but preserves current behavior. Tell me if you think we can just always pass the namespace to the store, disregarding whether or not it has a collection or not